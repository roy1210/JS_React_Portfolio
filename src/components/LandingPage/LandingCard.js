import React, { Component } from "react";
import profile from "../../img/profile.jpg";
import LandingRollDice from "./LandingRollDice";

class LandingCard extends Component {
  render() {
    return (
      <div id="Landing-f1_container">
        <div id="Landing-f1_card" className="Landing-shadow">
          <div class="front Landing-face">
            <img src={profile} alt="avatar" className="Landing-profile" />
          </div>
          <div className="back Landing-face center">
            <h3>You got me!</h3>
            <LandingRollDice />
          </div>
        </div>
      </div>
    );
  }
}

export default LandingCard;
