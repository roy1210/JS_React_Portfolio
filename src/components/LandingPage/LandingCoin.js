import React from "react";
import Bitcoin from "../../img/LandingPage/Bitcoin.png";
import DogCoin from "../../img/LandingPage/DogCoin.png";

const LandingCoin = ({ info, spinning }) => {
  let pic;
  if (info.side === "bitcoin") {
    pic = <img src={Bitcoin} alt={info.side} />;
  } else if (info.side === "dog-coin") {
    pic = <img src={DogCoin} alt={info.side} />;
  }

  return <div className={spinning ? "Coin-spinning" : "Coin"}>{pic}</div>;
};

export default LandingCoin;
