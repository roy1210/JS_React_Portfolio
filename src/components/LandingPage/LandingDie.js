import React, { Component } from "react";
import "../../CSS/Die.css";

class LandingDie extends Component {
  render() {
    return (
      <i
        className={`Die fas fa-dice-${this.props.face} ${this.props.rolling &&
          "shaking"}`}
      />
    );
  }
}

export default LandingDie;
