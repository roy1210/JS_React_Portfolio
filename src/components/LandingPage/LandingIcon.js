import React from 'react';

const LandingIcon = ({ url, icon }) => (
  <div className='Landing-social-links'>
    <a
      href={url}
      rel='noopener noreferrer'
      target='_blank'
      aria-label='Go to my social links'
    >
      <i className={icon} aria-hidden='true' />
    </a>
  </div>
);

export default LandingIcon;
