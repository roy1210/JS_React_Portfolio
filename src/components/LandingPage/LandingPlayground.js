import React, { Component } from "react";
import profile from "../../img/profile.jpg";
import LandingRollDice from "./LandingRollDice";

class LandingPlayground extends Component {
  render() {
    return (
      <div id="Landing-f1_container">
        <div id="Landing-f1_card" className="Landing-shadow">
          <div class="front Landing-face">
            <img src={profile} alt="avatar" className="Landing-profile" />
          </div>
          <div className="back center Landing-face">
            <p>Hidden game</p>
            <h3>You got me!</h3>
            <LandingRollDice />
          </div>
        </div>
      </div>
    );
  }
}

export default LandingPlayground;
