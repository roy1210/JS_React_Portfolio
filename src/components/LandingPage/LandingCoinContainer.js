import React, { Component } from "react";
import { choice } from "./LandingHelpers";
import LandingCoin from "./LandingCoin";
import Bitcoin from "../../img/Bitcoin.png";
import DogCoin from "../../img/DogCoin.png";
import "../../CSS/CoinContainer.css";

class LandingCoinContainer extends Component {
  static defaultProps = {
    coins: [{ side: "bitcoin" }, { side: "dog-coin" }]
  };
  constructor(props) {
    super(props);
    this.state = {
      currCoin: null,
      nFlips: 0,
      win: 0,
      lose: 0,
      spin: false,
      betOn: null
    };
  }

  flipCoin() {
    const newCoin = choice(this.props.coins);
    this.setState(st => {
      let newState = {
        ...st,
        currCoin: newCoin,
        nFlips: st.nFlips + 1,
        win: st.win + (newCoin.side === this.state.betOn ? 1 : 0),
        lose: st.lose + (newCoin.side !== this.state.betOn ? 1 : 0),
        spin: true
      };
      setTimeout(() => {
        this.setState({ spin: false });
      }, 500);
      return newState;
    });
  }

  async handleClick(e) {
    await this.setState(() => {
      return { betOn: e };
    });
    this.flipCoin();
  }

  render() {
    const mountPhoto = (
      <div style={{ display: "none" }}>
        <img src={Bitcoin} alt="bitcoin" />
        <img src={DogCoin} alt="dog-coin" />
      </div>
    );

    return (
      <div className="CoinContainer">
        <h2 className="CoinContainer-fade-in one">
          Flip a coin, challenge me!
        </h2>
        {this.state.currCoin && (
          <LandingCoin info={this.state.currCoin} spinning={this.state.spin} />
        )}
        <button
          onClick={this.handleClick.bind(this, "bitcoin")}
          disabled={this.state.spin}
        >
          {this.state.spin ? "Flipping..." : "Bet on Bitcoin"}
        </button>
        <button
          onClick={this.handleClick.bind(this, "dog-coin")}
          disabled={this.state.spin}
        >
          {this.state.spin ? "Flipping..." : "Bet on Dog-coin"}
        </button>
        {this.state.win > this.state.lose && (
          <div className="CoinContainer-win">
            <h2>
              Congrats, you have <span>won</span> {this.state.win} /{" "}
              {this.state.nFlips} times
            </h2>
          </div>
        )}
        {this.state.lose > this.state.win && (
          <div className="CoinContainer-lose">
            <h2>
              Oh no, you have <span>lost</span> {this.state.lose} /{" "}
              {this.state.nFlips} times
            </h2>
          </div>
        )}
        {this.state.lose === this.state.win && this.state.nFlips > 0 && (
          <div>
            <div className="CoinContainer-even">
              <h2>
                Now, we are <span>even</span>
              </h2>
            </div>
            <div className="CoinContainer-win">
              <h2>
                We have both <span>won</span> {this.state.win} /{" "}
                {this.state.nFlips} times
              </h2>
            </div>
          </div>
        )}
        <div className="CoinContainer-footer">
          <a
            href="https://codesandbox.io/s/flip-coin-6mu0b"
            rel="noopener noreferrer"
            target="_blank"
          >
            CodeSandBox
          </a>
        </div>
        {/* Mount photo to avoid lag. display: none */}
        {mountPhoto}
      </div>
    );
  }
}

export default LandingCoinContainer;
