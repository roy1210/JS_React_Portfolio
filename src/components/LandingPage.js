import React, { Component } from "react";
import { Grid, Cell } from "react-mdl";
import LandingPlayground from "./LandingPage/LandingPlayground";
import profile from "../img/profile.jpg";
import "../CSS/Landing.css";

class Landing extends Component {
  render() {
    return (
      <div
        style={{ width: "100%", margin: "auto" }}
        className="Landing-container"
      >
        <Grid className="Landing-grid">
          <Cell col={12} phone={12} tablet={12} className="Landing-SlideDown">
            <div className="Landing-phone-profile">
              <img src={profile} alt="profile" />
            </div>
            <LandingPlayground />
            <div className="Landing-banner-text">
              <h3>Hi, this is Roy. Nice to meet you.</h3>
              <h4>Front / Back - End Developer</h4>
              <hr />
              <p>JavaScript | React | Python | Golang | Solidity</p>
              <div className="Landing-social-links">
                {/* LinkedIn */}
                <a
                  href="https://www.linkedin.com/in/shoe-kure-2271ab90/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i className="fab fa-linkedin" aria-hidden="true" />
                </a>
                {/* Github */}
                <a
                  href="https://github.com/roy1210"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i className="fab fa-github" aria-hidden="true" />
                </a>
                {/* CodeSandBox */}
                <a
                  href="https://codesandbox.io/u/roy1210/sandboxes"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i className="fab fa-codepen" aria-hidden="true" />
                </a>
              </div>
            </div>
            <div className="LandingWrapper">
              <div className="LandingFooter">
                <small>Â© {new Date().getFullYear()} Shoe Kure</small>
              </div>
            </div>
          </Cell>
        </Grid>
      </div>
    );
  }
}

export default Landing;
