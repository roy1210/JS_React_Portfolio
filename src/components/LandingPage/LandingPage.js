import React, { Component } from 'react';
import { Grid, Cell } from 'react-mdl';
import LandingPlayground from './LandingPlayground';
import AppFooter from '../AppFooter';
import LandingIcon from './LandingIcon';
import Profile from '../../img/Profile.jpg';
import '../../CSS/Landing.css';

class LandingPage extends Component {
  // componentDidMount() {
  //   document.title = "Roy";
  // }
  render() {
    return (
      <div className='Landing-container'>
        <Grid className='Landing-grid'>
          <Cell col={12} phone={12} tablet={12}>
            <div className='Landing-SlideDown'>
              <div className='Landing-phone-profile'>
                <img src={Profile} alt='profile' />
              </div>
              <LandingPlayground />
              <div className='Landing-banner-text'>
                <h3>Hi, this is Roy. Nice to meet you.</h3>
                <h4>Web / Blockchain Developer</h4>
                <hr />
                <p>
                  <span className='Landing-javascript'>JavaScript</span> |{' '}
                  <span className='Landing-typescript'>TypeScript</span> |{' '}
                  <span className='Landing-react'>React</span> |{' '}
                  <span className='Landing-firebase'>Firebase</span> |{' '}
                  <span className='Landing-python'>Python</span> |{' '}
                  <span className='Landing-golang'>Golang</span> |{' '}
                  <span className='Landing-solidity'>Solidity</span>
                </p>
                <div className='Landing-social-links'>
                  <LandingIcon
                    url='https://www.linkedin.com/in/shoe-kure-2271ab90/'
                    iconClass='fa-linkedin'
                    iconName='faLinkedin'
                  />
                  <LandingIcon
                    url='https://github.com/roy1210'
                    iconClass='fa-github'
                    iconName='faGithub'
                  />
                  <LandingIcon
                    url='https://codesandbox.io/u/roy1210/sandboxes'
                    iconClass='fa-cube'
                    iconName='faCube'
                  />
                </div>
              </div>
            </div>
            <div className='Landing-marginBottom' />
            <div className='Landing-footer'>
              <AppFooter />
            </div>
          </Cell>
        </Grid>
      </div>
    );
  }
}

export default LandingPage;
