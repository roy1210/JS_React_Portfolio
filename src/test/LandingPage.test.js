import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from '../components/LandingPage/LandingPage';
import '../setupTests';

describe('About LandingPage', () => {
  it('renders without crashing', () => {
    const components = shallow(<LandingPage />);
    expect(components.debug()).toMatchSnapshot();
  });
});
