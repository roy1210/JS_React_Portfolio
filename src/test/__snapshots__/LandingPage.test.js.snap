// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About LandingPage renders without crashing 1`] = `
"<div className=\\"Landing-container\\">
  <Grid className=\\"Landing-grid\\">
    <Cell col={12} phone={12} tablet={12}>
      <div className=\\"Landing-SlideDown\\">
        <div className=\\"Landing-phone-profile\\">
          <img src=\\"Profile.jpg\\" alt=\\"profile\\" />
        </div>
        <LandingPlayground />
        <div className=\\"Landing-banner-text\\">
          <h3>
            Hi, this is Roy. Nice to meet you.
          </h3>
          <h4>
            Web / Blockchain Developer
          </h4>
          <hr />
          <p>
            <span className=\\"Landing-javascript\\">
              JavaScript
            </span>
             |
             
            <span className=\\"Landing-typescript\\">
              TypeScript
            </span>
             |
             
            <span className=\\"Landing-react\\">
              React
            </span>
             |
             
            <span className=\\"Landing-firebase\\">
              Firebase
            </span>
             |
             
            <span className=\\"Landing-python\\">
              Python
            </span>
             |
             
            <span className=\\"Landing-golang\\">
              Golang
            </span>
             |
             
            <span className=\\"Landing-solidity\\">
              Solidity
            </span>
          </p>
          <div className=\\"Landing-social-links\\">
            <LandingIcon url=\\"https://www.linkedin.com/in/shoe-kure-2271ab90/\\" icon=\\"fab fa-linkedin\\" />
            <LandingIcon url=\\"https://github.com/roy1210\\" icon=\\"fab fa-github\\" />
            <LandingIcon url=\\"https://codesandbox.io/u/roy1210/sandboxes\\" icon=\\"fab fas fa-cube\\" />
          </div>
        </div>
      </div>
      <div className=\\"Landing-marginBottom\\" />
      <div className=\\"Landing-footer\\">
        <Connect(AppFooter) />
      </div>
    </Cell>
  </Grid>
</div>"
`;
